src/Common/src/System/Net/CaseInsensitiveAscii.cs

```diff
@@ -85,46 +85,39 @@ public int Compare(object firstObject, object secondObject)
             return result;
         }
 
-        // ASCII string case insensitive hash function
-        private int FastGetHashCode(string myString)
-        {
-            int myHashCode = myString.Length;
-            if (myHashCode != 0)
-            {
-                myHashCode ^= AsciiToLower[(byte)myString[0]] << 24 ^ AsciiToLower[(byte)myString[myHashCode - 1]] << 16;
-            }
-            return myHashCode;
-        }
-
         // ASCII string case insensitive comparer
         public new bool Equals(object firstObject, object secondObject)
         {
             string firstString = firstObject as string;
             string secondString = secondObject as string;
-            if (firstString == null)
+            if ((object)firstString == (object)secondString)
             {
-                return secondString == null;
+                return true;
             }
-            if (secondString != null)
+
+            if (firstString == null | secondString == null)
```

Comment:
is `|` faster than `||` ?

Reply:
Generally that's a tough question because it's a matter of the gain of avoiding an operation over the loss of possible branch misprediction. Here though I don't think this would ever be called with nulls in the current uses, but we should still have it for robustness, so we're never not going to do the second comparison, so the potential gain in `||` doesn't exist, leaving only the potential loss.

Reply:
(That said, reference comparison with null is so cheap that generally avoiding the branch makes sense anyway. It puts the micro in micro-opt, but since one of `|` or `||` has to be done, I normally favour `|` for these.)